### ISSUE TITLE
What will happen in case of REVERT inside DELEGATE_CALL?

### ISSUE THREAD

**Issue by k06a** [Uniswap/contracts-vyper#39]
I see a lot of  `BAD JUMP DESTINATION` transactions here: https://etherscan.io/address/0x3958b4ec427f8fa24eb60f42821760e88d485f7f

SNX token calls `tokenFallback` on Uniswap contract in the middle of exchange and it seems Uniswap proxy handles this problem in a wrong way.

Proxy code is:
```
CALLDATASIZE
PUSH1 0x00
PUSH1 0x00
CALLDATACOPY
PUSH2 0x1000
PUSH1 0x00
CALLDATASIZE
PUSH1 0x00
PUSH20 0x2157a7894439191e520825fe9399ab8655e0f708
GAS
DELEGATE_CALL
ISZERO
PC
JUMPI
PUSH2 0x1000
PUSH1 0x00
RETURN
```

What does the latest part after `DELEGATE_CALL` means can anyone describe what should happen after `DELEGATE_CALL` reverts internally?

It seems Uniswap exchange called by another smart contract will eat all of its gas.

**Comment by haydenadams** [comment]
Yeah that comes from a vyper bug that wasn't discovered until after Uniswap launch

https://github.com/ethereum/vyper/issues/1091

**Comment by k06a** [comment]
So now we have an issue with Uniswap contract eating all the gas our smart contract provided.

**Comment by haydenadams** [comment]
@k06a on transaction fail, yes

**Comment by k06a** [comment]
@haydenadams found a reason and solution to 900K gas estimation to be dramatically decreased for SNX. Think could be delivered with upcoming token upgrade.
